textbook/2_grouping.qmd

```diff
@@ -0,0 +1,994 @@
+
+Here we use `color=region`:
+```{r fig.width=8, fig.height=6}
+library(readr)
+library(ggplot2)
+income_df <- read_csv('data/income_lifeexp_alcohol_1.csv', show_col_types = FALSE)
+
+ggplot(income_df, aes(
+    x=percentage_heavy_drinkers, 
+    y=life_expectancy_at_50, 
+    color=region
+)) + 
+geom_point()
+```
+
+:::
+
+Notice how you naturally interpret points of same color as clusters?
+
+But beware... The power of color fades when used irresponsibly. Using   colors that are too similar can blur the difference between groups. The more colors you include, the harder it is to stick to distinct hues.
+
+There is also the challenge of accessibility. Colorblindness affects approximately 8% of men and 0.5% of women globally. The most common variant is deuteranopia. This impairs the ability to distinguish between red and green hues, so designing your data visualization with this in mind is essential. While Altair’s default color palette is designed to be color-blind friendly, ggplot’s is not.
+
+We will learn much more about accessibility in chapter ...including specifics around which colors are safe to use together and which are suitable to represent different data types.
+
+### Shape
+
+Now our visualizations are really taking shape! Shape can be combined with color to provide an additional visual cue. Let's take a look at a classic example: the infamous traffic light. 
+
+Do you think that traffic lights are well designed? At first glance,  they break a major rules of accessibility, using the worst possible colors for people with deuteranopia (red and green). Thankfully, in many countries, traffic lights are encoded using both color and shape. A third encoding channel - position - is then used to establish predictable y-axis positions for both 'Stop' and 'Go' signals. 
+
+![](/img/traffic.png)
+
+Grouping by three encoding channels reinforces the visual contrast and has probably prevented many accidents.
+
+We can use the same syntax to encode both color and shape aesthetics: 
+
+::: {.panel-tabset}
+
+# Altair
+
+Here we use `shape='region'`:
+
+```{python}
+alt.Chart(income_df).mark_point().encode(
+    x='percentage_heavy_drinkers',
+    y='life_expectancy_at_50',
+    shape='region',
+    color='region',
+)
+```
+
+# ggplot
+
+Here we use `shape=region`:
+
+```{r fig.width=8, fig.height=6}
+ggplot(income_df, aes(
+    x=percentage_heavy_drinkers, 
+    y=life_expectancy_at_50, 
+    shape=region,
+    color=region
+)) + 
+geom_point()
+```
+
+:::
+
+While this combined approach makes grouping clearer, there are some pitfalls. For instance, line icons limit the total area that can be colored, reducing the impact of color. It is better to use solid shapes when combining shape and color.
+
+Additionally, size matters more than you'd think. Smaller shapes are difficult to distinguish. When plotting a dense scatterplot of small marks, try restricting the number of unique shapes to three or fewer, or consider avoiding using shape altogether.
```

Comment:
@joelostblom  I think I made this rule up... Visualization Analysis & Design by Tamara Munzner says that with 10x10 pixels you can make around 12 distinguishable shapes. Grouping by 12 shapes seems it could be a spaghetti nightmare